ingestion-beam/bin/download-document-sample

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+"""Download samples of json documents from the decoded and error stream.
+
+This is meant for integration testing, and is easily inspected through the command-line.
+For example, to count the total number documents per group:
+
+  cat document_sample.ndjson | \
+  jq -rc '.attributeMap | [.document_namespace, .document_type, .document_version]' | \
+  uniq -c
+"""
+
+import argparse
+import base64
+import gzip
+import json
+import logging
+import os
+import time
+
+from google.cloud import bigquery
+
+INGESTION_BEAM_ROOT = os.path.realpath(
+    os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+)
+
+# formatted using the BigQuery console formatter
+QUERY = """
+-- Create a PubSub compatible row with the most recent document samples that
+-- have been decoded.
+SELECT
+  STRUCT( document_namespace,
+    document_type,
+    document_version ) AS attributeMap,
+  payload
+FROM
+  `moz-fx-data-shared-prod`.monitoring.document_sample_nonprod_v1
+WHERE
+  document_decoded
+  AND submission_timestamp = (
```

Comment:
This looks like it's only going to return a single row. Is that what's intended?

Reply:
Each time the query is run, the result set is given a unique timestamp. So this is returning the last set of documents that have been sampled.

I also went through the entire workflow to enumerate the tables, and I've validated that the total number of documents in the dataset matches what I get locally.


```sql
select split(_TABLE_SUFFIX, '__')[offset(0)] as document_namespace, count(*)
from `amiyaguchi-dev.test_ingestion.*`
group by 1 order by 2
```

Row | document_namespace | f0_ |  
-- | -- | -- | --
1 | eng_workflow | 59 |  
2 | firefox_installer | 1000 |  
3 | mobile | 1000 |  
4 | firefox_launcher_process | 1000 |  
5 | pocket | 1000 |  
6 | org_mozilla_tv_firefox | 1011 |  
7 | org_mozilla_fenix_nightly | 1051 |  
8 | activity_stream | 3000 |  
9 | messaging_system | 3374 |  
10 | org_mozilla_fenix | 5000 |  
11 | telemetry | 24565 |  

```bash
jq '.attributeMap.document_namespace' document_sample.ndjson | uniq -c       
3000 "activity-stream"
  59 "eng-workflow"
1000 "firefox-installer"
1000 "firefox-launcher-process"
3374 "messaging-system"
1000 "mobile"
5000 "org-mozilla-fenix"
1051 "org-mozilla-fenix-nightly"
1011 "org-mozilla-tv-firefox"
1000 "pocket"
24565 "telemetry"
```